File: lib/java/src/main/java/uapi/internal/validation/ValidateString.java
package uapi.internal.validation;

import static uapi.internal.types.UString._STRING_NAME;
import static uapi.internal.validation.GetTypeUnexpectedValidationFailure.getTypeUnexpectedValidationFailure;

import java.util.List;

public class ValidateString {
    public static List<ValidationFailure> validateString(Object value) {
        if (value instanceof String) {
            return List.of();
        } else {
            return getTypeUnexpectedValidationFailure(List.of(), value, _STRING_NAME);
        }
    }
}

File: lib/ts/uapi/internal/validation/ValidateString.ts
import { getTypeUnexpectedValidationFailure } from 'uapi/internal/validation/GetTypeUnexpectedValidationFailure';
import { ValidationFailure } from 'uapi/internal/validation/ValidationFailure';
import { _STRING_NAME } from 'uapi/internal/types/UString';

export function validateString(value: any): ValidationFailure[] {
    if (typeof value === 'string') {
        return [];
    } else {
        return getTypeUnexpectedValidationFailure([], value, _STRING_NAME);
    }
}
